src/services/user.service.ts

```diff
@@ -0,0 +1,48 @@
+import { GuildMember, Role } from 'discord.js';
+import moment from 'moment';
+import { Maybe } from '../common/types';
+import { GuildService } from './guild.service';
+
+export class UserService {
+  private _STRIP_NON_NUMERIC: RegExp = /^\d/g;
+  private _ageThreshold = moment.duration(2, 'days');
+
+  constructor(private _guildService: GuildService) {}
+
+  /**
+   * Finds a user by tag, id, or username
+   */
+  public getMember(target: string): Maybe<GuildMember> {
+    const strippedID = target.replace(this._STRIP_NON_NUMERIC, ''); // Remove extra stuff that can come when an @
+
+    const user = this._guildService
+      .get()
+      .members.cache.filter((member) => {
+        const { nickname } = member;
+        const { tag, username, id } = member.user;
+        return [nickname, tag, username, id].some((t) => t === target || t === strippedID);
+      })
+      .first();
+
+    return user;
```

Comment:
You probably just inline this return

```suggestion
    return user = this._guildService
      .get()
      .members.cache.filter((member) => {
        const { nickname } = member;
        const { tag, username, id } = member.user;
        return [nickname, tag, username, id].some((t) => t === target || t === strippedID);
      })
      .first();
```